Remove ENABLE_DATALIST_ELEMENT directive

annevk opened the pull request:
#### 664399c555a59f2f0a5e041fa05ead22bb74531b
<pre>
Remove ENABLE_DATALIST_ELEMENT directive
<a href="https://bugs.webkit.org/show_bug.cgi?id=286140">https://bugs.webkit.org/show_bug.cgi?id=286140</a>

Reviewed by Aditya Keerthi.

And rely on the DataListElementEnabled preference instead. Add minimal
stubs to ports that currently do not implement this to make this
possible.

Rebase test expectations for WPE and Win and reorganize some
expectations such that the passing result is shared.

* LayoutTests/fast/forms/label/labelable-elements-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt:
* LayoutTests/platform/glib/fast/dom/Window/window-lookup-precedence-expected.txt: Renamed from LayoutTests/platform/gtk/fast/dom/Window/window-lookup-precedence-expected.txt.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/custom-elements/builtin-coverage-expected.txt: Renamed from LayoutTests/platform/gtk/imported/w3c/web-platform-tests/custom-elements/builtin-coverage-expected.txt.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/dom/nodes/Node-cloneNode-expected.txt: Renamed from LayoutTests/platform/gtk/imported/w3c/web-platform-tests/dom/nodes/Node-cloneNode-expected.txt.
* LayoutTests/platform/gtk/fast/forms/label/labelable-elements-expected.txt: Removed.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt: Removed.
* LayoutTests/platform/ios/fast/forms/label/labelable-elements-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt: Removed.
* LayoutTests/platform/mac-wk1/fast/forms/label/labelable-elements-expected.txt: Copied from LayoutTests/fast/forms/label/labelable-elements-expected.txt.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt.
* LayoutTests/platform/mac-wk2/fast/forms/label/labelable-elements-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt: Removed.
* LayoutTests/platform/win/imported/w3c/web-platform-tests/webIDL/ecmascript-binding/legacy-platform-object/DefineOwnProperty-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/WebIDL/ecmascript-binding/legacy-platform-object/DefineOwnProperty-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/constraints/form-validation-willValidate-datalist-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/hidden-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/input-list-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/text-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::hasDatalist const):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::toCSSValueID):
* Source/WebCore/css/html.css:
(input::-webkit-list-button):
(#if defined(ENABLE_DATALIST_ELEMENT) &amp;&amp; ENABLE_DATALIST_ELEMENT): Deleted.
* Source/WebCore/html/BaseDateAndTimeInputType.cpp:
(WebCore::BaseDateAndTimeInputType::shouldRespectListAttribute):
(WebCore::BaseDateAndTimeInputType::setupDateTimeChooserParameters):
* Source/WebCore/html/ColorInputType.cpp:
(WebCore::ColorInputType::suggestedColors const):
* Source/WebCore/html/DataListSuggestionInformation.h:
* Source/WebCore/html/HTMLDataListElement.cpp:
* Source/WebCore/html/HTMLDataListElement.h:
* Source/WebCore/html/HTMLDataListElement.idl:
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::dataListButtonElement const):
(WebCore::HTMLInputElement::listOptionValueAsDouble):
(WebCore::HTMLInputElement::attributeChanged):
(WebCore::HTMLInputElement::insertedIntoAncestor):
(WebCore::HTMLInputElement::removedFromAncestor):
(WebCore::ListAttributeTargetObserver::idTargetChanged):
* Source/WebCore/html/HTMLInputElement.h:
* Source/WebCore/html/HTMLInputElement.idl:
* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::attributeChanged):
(WebCore::HTMLOptionElement::childrenChanged):
* Source/WebCore/html/HTMLTagNames.in:
* Source/WebCore/html/InputType.cpp:
(WebCore::InputType::findClosestTickMarkValue):
(WebCore::InputType::themeSupportsDataListUI): Deleted.
(WebCore::InputType::shouldRespectListAttribute): Deleted.
(WebCore::InputType::dataListMayHaveChanged): Deleted.
* Source/WebCore/html/InputType.h:
(WebCore::InputType::dataListButtonElement const):
(WebCore::InputType::shouldRespectListAttribute):
(WebCore::InputType::isFocusingWithDataListDropdown const):
(WebCore::InputType::dataListMayHaveChanged):
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::shouldRespectListAttribute):
(WebCore::RangeInputType::findClosestTickMarkValue):
* Source/WebCore/html/RangeInputType.h:
* Source/WebCore/html/TextFieldInputType.cpp:
(WebCore::TextFieldInputType::~TextFieldInputType):
(WebCore::TextFieldInputType::showPicker):
(WebCore::TextFieldInputType::handleKeydownEvent):
(WebCore::TextFieldInputType::handleKeydownEventForSpinButton):
(WebCore::TextFieldInputType::elementDidBlur):
(WebCore::TextFieldInputType::createShadowSubtree):
(WebCore::TextFieldInputType::removeShadowSubtree):
(WebCore::TextFieldInputType::shouldRespectListAttribute):
(WebCore::TextFieldInputType::didSetValueByUserEdit):
* Source/WebCore/html/TextFieldInputType.h:
* Source/WebCore/html/ValidatedFormListedElement.cpp:
(WebCore::ValidatedFormListedElement::computeWillValidate const):
* Source/WebCore/html/shadow/DataListButtonElement.cpp:
* Source/WebCore/html/shadow/DataListButtonElement.h:
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::RenderSliderContainer::computeLogicalHeight const):
(WebCore::SliderThumbElement::setPositionFromPoint):
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::EmptyChromeClient::createColorChooser): Deleted.
(WebCore::EmptyChromeClient::createDataListSuggestionPicker): Deleted.
* Source/WebCore/loader/EmptyClients.h:
* Source/WebCore/page/Chrome.cpp:
* Source/WebCore/page/Chrome.h:
* Source/WebCore/page/ChromeClient.h:
* Source/WebCore/platform/DataListSuggestionPicker.h:
* Source/WebCore/platform/DataListSuggestionsClient.h:
* Source/WebCore/platform/StyleAppearance.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/StyleAppearance.h:
* Source/WebCore/platform/graphics/adwaita/SliderTrackAdwaita.cpp:
(WebCore::SliderTrackAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/TextFieldAdwaita.cpp:
(WebCore::TextFieldAdwaita::draw):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.cpp:
(WebCore::SliderTrackPart::drawTicks const):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.h:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.h:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.mm:
(WebCore::ControlMac::drawListButton):
* Source/WebCore/platform/graphics/mac/controls/SearchFieldMac.mm:
(WebCore::SearchFieldMac::draw):
* Source/WebCore/platform/graphics/mac/controls/SliderTrackMac.mm:
(WebCore::SliderTrackMac::draw):
* Source/WebCore/platform/graphics/mac/controls/TextFieldMac.mm:
(WebCore::TextFieldMac::draw):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustStyle):
(WebCore::RenderTheme::autoAppearanceForElement const):
(WebCore::updateSliderTrackPartForRenderer):
(WebCore::RenderTheme::createControlPart const):
(WebCore::RenderTheme::extractControlStyleStatesForRendererInternal const):
(WebCore::RenderTheme::paint):
(WebCore::RenderTheme::hasListButtonPressed const):
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::sliderTickSize const):
(WebCore::RenderTheme::sliderTickOffsetFromTrackCenter const):
(WebCore::RenderTheme::paintListButton):
(WebCore::RenderTheme::supportsDataListUI const): Deleted.
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::adjustListButtonStyle const):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.h:
* Source/WebCore/rendering/ios/RenderThemeIOS.h:
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::isControlStyled const):
(WebCore::RenderThemeIOS::paintSliderTrack):
(WebCore::RenderThemeIOS::sliderTickOffsetFromTrackCenter const):
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::sliderTickOffsetFromTrackCenter const):
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:
* Source/WebKit/Shared/FocusedElementInformation.h:
* Source/WebKit/Shared/FocusedElementInformation.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/ios/InteractionInformationAtPosition.h:
* Source/WebKit/Shared/ios/InteractionInformationAtPosition.serialization.in:
* Source/WebKit/UIProcess/API/gtk/PageClientImpl.cpp:
(WebKit::PageClientImpl::createDataListSuggestionsDropdown):
* Source/WebKit/UIProcess/API/gtk/PageClientImpl.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewTestingIOS.mm:
(-[WKWebView _selectDataListOption:]):
(-[WKWebView _isShowingDataListSuggestions]):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::createDataListSuggestionsDropdown):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.h:
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebDataListSuggestionsDropdown.cpp:
* Source/WebKit/UIProcess/WebDataListSuggestionsDropdown.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::showDataListSuggestions):
(WebKit::WebPageProxy::closeOverlayedViews):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/gtk/WebDataListSuggestionsDropdownGtk.cpp:
* Source/WebKit/UIProcess/gtk/WebDataListSuggestionsDropdownGtk.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::createDataListSuggestionsDropdown):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView setUpInteraction]):
(-[WKContentView cleanUpInteraction]):
(-[WKContentView _endEditing]):
(-[WKContentView inputViewForWebView]):
(-[WKContentView textInteractionGesture:shouldBeginAtPoint:]):
(-[WKContentView insertTextSuggestion:]):
(-[WKContentView _elementDidBlur]):
(-[WKContentView updateTextSuggestionsForInputDelegate]):
(-[WKContentView isShowingDataListSuggestions]):
* Source/WebKit/UIProcess/ios/WebDataListSuggestionsDropdownIOS.h:
* Source/WebKit/UIProcess/ios/WebDataListSuggestionsDropdownIOS.mm:
* Source/WebKit/UIProcess/ios/forms/WKFormColorControl.mm:
(-[WKColorPicker focusedElementSuggestedColors]):
(-[WKColorPicker updateColorPickerState]):
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::createDataListSuggestionsDropdown):
* Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.h:
* Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.mm:
* Source/WebKit/UIProcess/playstation/PageClientImpl.cpp:
(WebKit::PageClientImpl::createColorPicker): Deleted.
* Source/WebKit/UIProcess/playstation/PageClientImpl.h:
* Source/WebKit/UIProcess/win/PageClientImpl.cpp:
(WebKit::PageClientImpl::createColorPicker): Deleted.
* Source/WebKit/UIProcess/win/PageClientImpl.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::containerElementForElement):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebDataListSuggestionPicker.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebDataListSuggestionPicker.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::textInteractionPositionInformation):
(WebKit::WebPage::positionInformation):
(WebKit::WebPage::focusedElementInformation):
* Source/WebKitLegacy/mac/DOM/DOMHTMLInputElement.mm:
(-[DOMHTMLInputElement list]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
(WebChromeClient::createDataListSuggestionPicker):
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsMac.cmake:
* Source/cmake/OptionsWin.cmake:
* Source/cmake/WebKitFeatures.cmake:
* Tools/Scripts/webkitperl/FeatureList.pm:
* Tools/TestWebKitAPI/Tests/ios/DataListTextSuggestionTests.mm:

Canonical link: <a href="https://commits.webkit.org/289182@main">https://commits.webkit.org/289182@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2ed4d1ae24775fd59270891f6a0548ac6eaf231e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/85655 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/5343 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/40055 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/90739 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/36650 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/87700 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/5559 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/13325 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/66562 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24371 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/88664 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/4278 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/77775 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46845 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4142 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/32038 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/35723 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/78661 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/74796 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/32894 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/92418 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/84648 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/12974 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/9529 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/75298 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/13187 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/73607 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/74437 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/18659 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/17104 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/5084 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13349 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/12993 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/18368 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/107039 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/12788 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/25801 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/16220 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/14574 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2412,19 +2412,18 @@ DataDetectorTypes:
     WebKit:
-      "PLATFORM(WATCHOS)": false
-      default: true
+      "PLATFORM(COCOA) && !PLATFORM(WATCHOS) || PLATFORM(GTK)": true
```
For readability?

```suggestion
      "(PLATFORM(COCOA) && !PLATFORM(WATCHOS)) || PLATFORM(GTK)": true
```

pxlcoder commented on Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
```diff
@@ -2412,19 +2412,18 @@ DataDetectorTypes:
     WebCore:
-      "PLATFORM(WATCHOS)": false
-      default: true
+      "PLATFORM(COCOA) && !PLATFORM(WATCHOS) || PLATFORM(GTK)": true
```
Ditto.

pxlcoder commented on Source/WebCore/html/HTMLInputElement.idl:
```diff
@@ -59,7 +59,7 @@ enum AutofillButtonType {
     [CEReactions=NotNeeded] attribute unsigned long height;
     attribute boolean indeterminate;
-    [Conditional=DATALIST_ELEMENT] readonly attribute HTMLElement list;
+    readonly attribute HTMLElement list;
```
Strange that this never had an `EnabledBySetting`, but not new.

annevk replied:
Yeah, I had added that initially but that regressed WK1 tests and I don't really want to risk that as part of this patch. Also not entirely sure it's worth it as this attribute (and corresponding element) should really be supported by all ports.

annevk committed:
Remove ENABLE_DATALIST_ELEMENT directive
https://bugs.webkit.org/show_bug.cgi?id=286140

Reviewed by Aditya Keerthi.

And rely on the DataListElementEnabled preference instead. Add minimal
stubs to ports that currently do not implement this to make this
possible.

Rebase test expectations for WPE and Win and reorganize some
expectations such that the passing result is shared.

* LayoutTests/fast/forms/label/labelable-elements-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt:
* LayoutTests/platform/glib/fast/dom/Window/window-lookup-precedence-expected.txt: Renamed from LayoutTests/platform/gtk/fast/dom/Window/window-lookup-precedence-expected.txt.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/custom-elements/builtin-coverage-expected.txt: Renamed from LayoutTests/platform/gtk/imported/w3c/web-platform-tests/custom-elements/builtin-coverage-expected.txt.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/dom/nodes/Node-cloneNode-expected.txt: Renamed from LayoutTests/platform/gtk/imported/w3c/web-platform-tests/dom/nodes/Node-cloneNode-expected.txt.
* LayoutTests/platform/gtk/fast/forms/label/labelable-elements-expected.txt: Removed.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt: Removed.
* LayoutTests/platform/ios/fast/forms/label/labelable-elements-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt: Removed.
* LayoutTests/platform/mac-wk1/fast/forms/label/labelable-elements-expected.txt: Copied from LayoutTests/fast/forms/label/labelable-elements-expected.txt.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt: Copied from LayoutTests/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt.
* LayoutTests/platform/mac-wk2/fast/forms/label/labelable-elements-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/html/semantics/interfaces-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt: Removed.
* LayoutTests/platform/win/imported/w3c/web-platform-tests/webIDL/ecmascript-binding/legacy-platform-object/DefineOwnProperty-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/WebIDL/ecmascript-binding/legacy-platform-object/DefineOwnProperty-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/constraints/form-validation-willValidate-datalist-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/hidden-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/input-list-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/text-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/shadow-dom/input-element-list-expected.txt: Removed.
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::hasDatalist const):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::toCSSValueID):
* Source/WebCore/css/html.css:
(input::-webkit-list-button):
(#if defined(ENABLE_DATALIST_ELEMENT) && ENABLE_DATALIST_ELEMENT): Deleted.
* Source/WebCore/html/BaseDateAndTimeInputType.cpp:
(WebCore::BaseDateAndTimeInputType::shouldRespectListAttribute):
(WebCore::BaseDateAndTimeInputType::setupDateTimeChooserParameters):
* Source/WebCore/html/ColorInputType.cpp:
(WebCore::ColorInputType::suggestedColors const):
* Source/WebCore/html/DataListSuggestionInformation.h:
* Source/WebCore/html/HTMLDataListElement.cpp:
* Source/WebCore/html/HTMLDataListElement.h:
* Source/WebCore/html/HTMLDataListElement.idl:
* Source/WebCore/html/HTMLInputElement.cpp:
(WebCore::HTMLInputElement::dataListButtonElement const):
(WebCore::HTMLInputElement::listOptionValueAsDouble):
(WebCore::HTMLInputElement::attributeChanged):
(WebCore::HTMLInputElement::insertedIntoAncestor):
(WebCore::HTMLInputElement::removedFromAncestor):
(WebCore::ListAttributeTargetObserver::idTargetChanged):
* Source/WebCore/html/HTMLInputElement.h:
* Source/WebCore/html/HTMLInputElement.idl:
* Source/WebCore/html/HTMLOptionElement.cpp:
(WebCore::HTMLOptionElement::attributeChanged):
(WebCore::HTMLOptionElement::childrenChanged):
* Source/WebCore/html/HTMLTagNames.in:
* Source/WebCore/html/InputType.cpp:
(WebCore::InputType::findClosestTickMarkValue):
(WebCore::InputType::themeSupportsDataListUI): Deleted.
(WebCore::InputType::shouldRespectListAttribute): Deleted.
(WebCore::InputType::dataListMayHaveChanged): Deleted.
* Source/WebCore/html/InputType.h:
(WebCore::InputType::dataListButtonElement const):
(WebCore::InputType::shouldRespectListAttribute):
(WebCore::InputType::isFocusingWithDataListDropdown const):
(WebCore::InputType::dataListMayHaveChanged):
* Source/WebCore/html/RangeInputType.cpp:
(WebCore::RangeInputType::shouldRespectListAttribute):
(WebCore::RangeInputType::findClosestTickMarkValue):
* Source/WebCore/html/RangeInputType.h:
* Source/WebCore/html/TextFieldInputType.cpp:
(WebCore::TextFieldInputType::~TextFieldInputType):
(WebCore::TextFieldInputType::showPicker):
(WebCore::TextFieldInputType::handleKeydownEvent):
(WebCore::TextFieldInputType::handleKeydownEventForSpinButton):
(WebCore::TextFieldInputType::elementDidBlur):
(WebCore::TextFieldInputType::createShadowSubtree):
(WebCore::TextFieldInputType::removeShadowSubtree):
(WebCore::TextFieldInputType::shouldRespectListAttribute):
(WebCore::TextFieldInputType::didSetValueByUserEdit):
* Source/WebCore/html/TextFieldInputType.h:
* Source/WebCore/html/ValidatedFormListedElement.cpp:
(WebCore::ValidatedFormListedElement::computeWillValidate const):
* Source/WebCore/html/shadow/DataListButtonElement.cpp:
* Source/WebCore/html/shadow/DataListButtonElement.h:
* Source/WebCore/html/shadow/SliderThumbElement.cpp:
(WebCore::RenderSliderContainer::computeLogicalHeight const):
(WebCore::SliderThumbElement::setPositionFromPoint):
* Source/WebCore/loader/EmptyClients.cpp:
(WebCore::EmptyChromeClient::createColorChooser): Deleted.
(WebCore::EmptyChromeClient::createDataListSuggestionPicker): Deleted.
* Source/WebCore/loader/EmptyClients.h:
* Source/WebCore/page/Chrome.cpp:
* Source/WebCore/page/Chrome.h:
* Source/WebCore/page/ChromeClient.h:
* Source/WebCore/platform/DataListSuggestionPicker.h:
* Source/WebCore/platform/DataListSuggestionsClient.h:
* Source/WebCore/platform/StyleAppearance.cpp:
(WebCore::operator<<):
* Source/WebCore/platform/StyleAppearance.h:
* Source/WebCore/platform/graphics/adwaita/SliderTrackAdwaita.cpp:
(WebCore::SliderTrackAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/TextFieldAdwaita.cpp:
(WebCore::TextFieldAdwaita::draw):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.cpp:
(WebCore::SliderTrackPart::drawTicks const):
* Source/WebCore/platform/graphics/controls/SliderTrackPart.h:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.h:
* Source/WebCore/platform/graphics/mac/controls/ControlMac.mm:
(WebCore::ControlMac::drawListButton):
* Source/WebCore/platform/graphics/mac/controls/SearchFieldMac.mm:
(WebCore::SearchFieldMac::draw):
* Source/WebCore/platform/graphics/mac/controls/SliderTrackMac.mm:
(WebCore::SliderTrackMac::draw):
* Source/WebCore/platform/graphics/mac/controls/TextFieldMac.mm:
(WebCore::TextFieldMac::draw):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::adjustStyle):
(WebCore::RenderTheme::autoAppearanceForElement const):
(WebCore::updateSliderTrackPartForRenderer):
(WebCore::RenderTheme::createControlPart const):
(WebCore::RenderTheme::extractControlStyleStatesForRendererInternal const):
(WebCore::RenderTheme::paint):
(WebCore::RenderTheme::hasListButtonPressed const):
* Source/WebCore/rendering/RenderTheme.h:
(WebCore::RenderTheme::sliderTickSize const):
(WebCore::RenderTheme::sliderTickOffsetFromTrackCenter const):
(WebCore::RenderTheme::paintListButton):
(WebCore::RenderTheme::supportsDataListUI const): Deleted.
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::adjustListButtonStyle const):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.h:
* Source/WebCore/rendering/ios/RenderThemeIOS.h:
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
(WebCore::RenderThemeIOS::isControlStyled const):
(WebCore::RenderThemeIOS::paintSliderTrack):
(WebCore::RenderThemeIOS::sliderTickOffsetFromTrackCenter const):
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::RenderThemeMac::sliderTickOffsetFromTrackCenter const):
* Source/WebCore/style/InspectorCSSOMWrappers.cpp:
(WebCore::Style::InspectorCSSOMWrappers::collectDocumentWrappers):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):
* Source/WebCore/style/UserAgentStyle.h:
* Source/WebKit/Shared/FocusedElementInformation.h:
* Source/WebKit/Shared/FocusedElementInformation.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/ios/InteractionInformationAtPosition.h:
* Source/WebKit/Shared/ios/InteractionInformationAtPosition.serialization.in:
* Source/WebKit/UIProcess/API/gtk/PageClientImpl.cpp:
(WebKit::PageClientImpl::createDataListSuggestionsDropdown):
* Source/WebKit/UIProcess/API/gtk/PageClientImpl.h:
* Source/WebKit/UIProcess/API/ios/WKWebViewTestingIOS.mm:
(-[WKWebView _selectDataListOption:]):
(-[WKWebView _isShowingDataListSuggestions]):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.cpp:
(WebKit::PageClientImpl::createDataListSuggestionsDropdown):
* Source/WebKit/UIProcess/API/wpe/PageClientImpl.h:
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebDataListSuggestionsDropdown.cpp:
* Source/WebKit/UIProcess/WebDataListSuggestionsDropdown.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::showDataListSuggestions):
(WebKit::WebPageProxy::closeOverlayedViews):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/WebPageProxyInternals.h:
* Source/WebKit/UIProcess/gtk/WebDataListSuggestionsDropdownGtk.cpp:
* Source/WebKit/UIProcess/gtk/WebDataListSuggestionsDropdownGtk.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::createDataListSuggestionsDropdown):
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView setUpInteraction]):
(-[WKContentView cleanUpInteraction]):
(-[WKContentView _endEditing]):
(-[WKContentView inputViewForWebView]):
(-[WKContentView textInteractionGesture:shouldBeginAtPoint:]):
(-[WKContentView insertTextSuggestion:]):
(-[WKContentView _elementDidBlur]):
(-[WKContentView updateTextSuggestionsForInputDelegate]):
(-[WKContentView isShowingDataListSuggestions]):
* Source/WebKit/UIProcess/ios/WebDataListSuggestionsDropdownIOS.h:
* Source/WebKit/UIProcess/ios/WebDataListSuggestionsDropdownIOS.mm:
* Source/WebKit/UIProcess/ios/forms/WKFormColorControl.mm:
(-[WKColorPicker focusedElementSuggestedColors]):
(-[WKColorPicker updateColorPickerState]):
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::createDataListSuggestionsDropdown):
* Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.h:
* Source/WebKit/UIProcess/mac/WebDataListSuggestionsDropdownMac.mm:
* Source/WebKit/UIProcess/playstation/PageClientImpl.cpp:
(WebKit::PageClientImpl::createColorPicker): Deleted.
* Source/WebKit/UIProcess/playstation/PageClientImpl.h:
* Source/WebKit/UIProcess/win/PageClientImpl.cpp:
(WebKit::PageClientImpl::createColorPicker): Deleted.
* Source/WebKit/UIProcess/win/PageClientImpl.h:
* Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp:
(WebKit::containerElementForElement):
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebDataListSuggestionPicker.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebDataListSuggestionPicker.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm:
(WebKit::textInteractionPositionInformation):
(WebKit::WebPage::positionInformation):
(WebKit::WebPage::focusedElementInformation):
* Source/WebKitLegacy/mac/DOM/DOMHTMLInputElement.mm:
(-[DOMHTMLInputElement list]):
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebChromeClient.mm:
(WebChromeClient::createDataListSuggestionPicker):
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsMac.cmake:
* Source/cmake/OptionsWin.cmake:
* Source/cmake/WebKitFeatures.cmake:
* Tools/Scripts/webkitperl/FeatureList.pm:
* Tools/TestWebKitAPI/Tests/ios/DataListTextSuggestionTests.mm:

Canonical link: https://commits.webkit.org/289182@main
